.github/workflows/Unit-Test.yml

```diff
@@ -3,84 +3,76 @@ name: Run Unit Tests on PR
-            echo "PR #${PR_NUMBER} has these ARMv7L compatible packages: ${ARMV7L_PACKAGES}" && \
-            if [[ $GLIBC_237_COMPATIBLE_PACKAGES ]] ; then x86_64_CONTAINER="satmandu/crewbuild:hatch-x86_64.m126"; else x86_64_CONTAINER="satmandu/crewbuild:nocturne-x86_64.m90"; fi &&\
-            export container_cmdline="/usr/bin/sudo -u chronos LD_LIBRARY_PATH=/usr/local/lib64:/lib64 /usr/local/bin/bash -c \"export ALL_CHANGED_FILES='${ALL_CHANGED_FILES}' && \
-            export CHANGED_PACKAGES='${CHANGED_PACKAGES}' && \
-            echo 'ALL_CHANGED FILES: ${ALL_CHANGED_FILES}.' && \
-            echo 'CHANGED PACKAGES: ${CHANGED_PACKAGES}.' && \
-            echo 'CREW_REPO is ${{ github.event.pull_request.head.repo.clone_url }}' && \
-            echo 'CREW_BRANCH is ${{ github.head_ref }}' && \
-            CREW_REPO=${{ github.event.pull_request.head.repo.clone_url }} CREW_BRANCH=${{ github.head_ref }} crew update && \
-            /usr/local/lib/crew/tests/unit_test.sh\"" && \
-            echo $container_cmdline && \
-            sudo docker run --rm \
-            --platform linux/amd64 \
-            -i ${x86_64_CONTAINER} \
-            /bin/bash -c "$container_cmdline"
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3.2.0
-        # This is not the best way to do things, a matrix would certainly be better.
-      - name: Unit Tests (armv7l)
+            # Convert "packages/foo.rb packages/bar.rb" (from steps.changed-ruby-files.outputs.all_changed_files) into "foo bar"
+            echo "CHANGED_PACKAGES=$(echo "${{ steps.changed-ruby-files.outputs.all_changed_files }}" | xargs basename -s .rb | xargs)" >> $GITHUB_ENV
+      - name: Determine glibc and architecture package compatibility
+        run: |
+            # If a package doesnt have a min_glibc value, or if its below 2.27, add it to the list
+            echo "PR #${{ github.event.pull_request.number }} has these possibly Glibc 2.27 compatible packages: $(for i in ${CHANGED_PACKAGES} ; do if ! grep -q min_glibc packages/$i.rb; then echo $i ; else grep min_glibc packages/$i.rb | tr -d \' | awk '{exit $2 <= 2.27}' || echo $i ; fi ; done | xargs)"
+
+            # If a package doesnt have a min_glibc value, or if its below 2.37, add it to GLIBC_237_COMPATIBLE_PACKAGES
+            echo "GLIBC_237_COMPATIBLE_PACKAGES=$(for i in ${CHANGED_PACKAGES} ; do grep min_glibc packages/$i.rb | tr -d \' | awk '{exit $2 <= 2.37}' || echo $i ; done | xargs)" >> $GITHUB_ENV
+            echo "PR #${{ github.event.pull_request.number }} has these possibly Glibc 2.37 compatible packages: ${GLIBC_237_COMPATIBLE_PACKAGES}"
+
+            # If a package has a compatibility of 'all' or one that includes 'x86_64', add it to the list.
+            echo "PR #${{ github.event.pull_request.number }} has these x86_64 compatible packages: $(for p in ${CHANGED_PACKAGES}; do grep -q "compatibility 'all'\|compatibility '.*x86_64.*'" packages/${p}.rb && echo ${p}; done | xargs)"
```

Comment:
What variable is this going into?

Reply:
its not going into a variable, because we didn't actually use the variable before aside from printing it so there's no point to creating one.